aea/mail/base.py

```diff
@@ -392,39 +369,39 @@ def __init__(
                 bool(protocol_id) or bool(protocol_specification_id),
```

Comment:
Ideally, we allow `bytes` to only (!) be supplied with the protocol_specification_id. That is safer!